### PR TITLE
optionalSliderInput - out of range

### LINKED ISSUE #88: [Bug]: optionalSliderInput warns when value is length 2
### What happened?

When specifying `value` as a 2 element numeric vector, `optionalSliderInput` throws a warning. This arises from [comparing `value` to `min` & `max`](https://github.com/insightsengineering/teal.widgets/blob/d253b9280c2699ece22f4a3c34ca2aea518cb980/R/optionalInput.R#L374). R doesn't like comparing a length 2 vector with a length 1.

```
> library(teal.widgets)
> optionalSliderInput(inputId = 'test', label = 'test', min = 0, max = 1, value = c(0,1))
<div class="form-group shiny-input-container">
  <label class="control-label" id="test-label" for="test">test</label>
  <input class="js-range-slider" id="test" data-skin="shiny" data-type="double" data-min="0" data-max="1" data-from="0" data-to="1" data-step="0.01" data-grid="true" data-grid-num="10" data-grid-snap="false" data-prettify-separator="," data-prettify-enabled="true" data-keyboard="true" data-drag-interval="true" data-data-type="number"/>
</div>
Warning messages:
1: In min > value || max < value :
  'length(x) = 2 > 1' in coercion to 'logical(1)'
2: In min > value || max < value :
  'length(x) = 2 > 1' in coercion to 'logical(1)'
```

The expected output is the returned html string with no warnings.

### sessionInfo()

```R
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] teal.widgets_0.1.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9       digest_0.6.29    later_1.3.0      mime_0.12        R6_2.5.1        
 [6] jsonlite_1.8.0   lifecycle_1.0.2  xtable_1.8-4     magrittr_2.0.3   cachem_1.0.6    
[11] rlang_1.0.5      cli_3.4.0        rstudioapi_0.13  promises_1.2.0.1 jquerylib_0.1.4 
[16] bslib_0.4.0      ellipsis_0.3.2   tools_4.2.1      shiny_1.7.2      httpuv_1.6.6    
[21] fastmap_1.1.0    compiler_4.2.1   memoise_2.0.1    htmltools_0.5.3  sass_0.4.2
```

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by Polkas**
closes #88 

Fixed the  optionalSliderInput conditional which assumed only single value for the slider position.
Another fix are proper values when min and max are NA.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 ### Bug fixes
 * Fixed bug in `verbatim_popup_srv` where `disabled` argument didn't influence the popup button. 
+* Fixed `optionalSliderInput` validation regarding the possible range.
```
Comment (nikolas-burkoff) [inline NEWS.md:11]:
```suggestion
* Fixed `optionalSliderInput` validation to remove a warning message in certain cases.
```

File: R/optionalInput.R [hunk 2]
```diff
@@ -367,11 +367,23 @@ extract_raw_choices <- function(choices, sep) {
 #' @examples
 #' optionalSliderInput("a", "b", 0, 1, 0.2)
 optionalSliderInput <- function(inputId, label, min, max, value, label_help = NULL, ...) { # nolint
+  checkmate::assert_number(min)
+  checkmate::assert_number(max)
+  checkmate::assert_numeric(value, min.len = 1, max.len = 2)
```
Comment (nikolas-burkoff) [inline R/optionalInput.R:372]:
```suggestion
  checkmate::assert_numeric(value, min.len = 1, max.len = 2, any.missing = FALSE)
```

File: R/optionalInput.R [hunk 3]
```diff
@@ -367,11 +367,23 @@ extract_raw_choices <- function(choices, sep) {
...
+  } else {
+    value1 <- value
+    value2 <- value
+  }
+
   hide <- if (is.na(min) || is.na(max)) {
-    min <- value - 1
-    max <- value + 1
+    min <- value1 - 1
+    max <- value2 + 1
```
Comment (nikolas-burkoff) [inline R/optionalInput.R:384]: This feels odd - what if min is NA but max is not...

**Review by nikolas-burkoff — COMMENTED** [review]
A couple of comments but looks good

File: R/optionalInput.R (hunk 3 above)
Comment (Polkas) [reply R/optionalInput.R]: i fix that

File: R/optionalInput.R [hunk 4]
```diff
@@ -367,16 +367,38 @@ extract_raw_choices <- function(choices, sep) {
...
+  is_na_min <- is.na(min)
+  is_na_max <- is.na(max)
+
+  hide <- if (is_na_min || is_na_max) {
     TRUE
-  } else if (min > value || max < value) {
-    stop("arguments inconsistent: min <= value <= max expected")
   } else {
     FALSE
   }
```
Comment (nikolas-burkoff) [inline R/optionalInput.R:381]: surely just `hide <- is_na_min || is_na_max` ?

**Review by nikolas-burkoff — APPROVED** [review]
one small comment but looks good

File: R/optionalInput.R (hunk 4 above)
Comment (Polkas) [reply R/optionalInput.R]: agree 2 lines less

**Pull request merged**